firebase-firestore/src/main/java/com/google/firebase/firestore/model/mutation/ArrayTransformOperation.java

```diff
@@ -133,4 +156,44 @@ protected Value apply(@Nullable Value previousValue) {
       return Value.newBuilder().setArrayValue(result).build();
     }
   }
+
+  /**
+   * An array transform operation that delegate its work to an internal list of other array
+   * transform operations.
+   *
+   * <p>When a list of array transform operations being applied, the order of application needs to
+   * be preserved, and they are not be squashed into one simple mutation. This class exists such
+   * that the squashed array transform operation still has the same order.
+   */
+  static class ArrayTransformList extends ArrayTransformOperation {
```

Comment:
I feel there might be better name if you can fit one of the patterns here - https://en.wikipedia.org/wiki/Software_design_pattern